Sources/StreamChatUI/Utils/ImageCDN.swift

```diff
@@ -0,0 +1,39 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+public var cachingKeyProvider: CachingKeyProvider = StreamCachingKeyProvider()
+
+/// Caching key provider is providing custimised (filtered) key to be used for caching.
+///
+/// The URL can contain extra parameters that are changing every time. Those parameters are effectively blocking the caching.
+/// By providing customised (filtered) key, the cache will work as expected.
+public protocol CachingKeyProvider {
+    /// Customised (filtered) key for image cache.
+    /// - Parameter imageURL: URL of the image that should be customised (filtered).
+    /// - Returns: String to be used as an image cache key.
+    func cachingKey(forImage url: URL) -> String
+}
+
+open class StreamCachingKeyProvider: CachingKeyProvider {
+    public static var streamCDNURL = "stream-io-cdn.com"
+    
+    public func cachingKey(forImage url: URL) -> String {
```

Comment:
I'd step out from inheritance and make the `StreamCachingKeyProvider` a struct. We have a protocol and customers will build their own implementation of `CachingKeyProvider` from scratch anyway

Reply:
You are right, good catch!